app/models/cfe_result.rb

```diff
@@ -6,7 +6,11 @@ def initialize(api_response)
   end
 
   def ineligible_gross_income?
-    api_response.dig(:result_summary, :gross_income, :proceeding_types).first[:result] == "ineligible"
+    early_gross_income_result == "ineligible"
+  end
+
+  def early_gross_income_result
+    api_response.dig(:result_summary, :gross_income, :proceeding_types, 0, :result)
```

Comment:
how are we assuring ourselves that this is indeed an early result? This dig would return a value at any time it is called.

Reply:
yeah you are right, it is badly named because it is actually just a gross income result!

Reply:
renamed this method